cellprofiler/modules/measureobjectsizeshape.py

```diff
@@ -285,120 +287,192 @@ def run(self, workspace):
+                ff = 4.0 * np.pi * objects.areas / mperimeters ** 2
                 #
-                # Zernike features
+                # Feret diameter
                 #
-                zf_l = cpmz.zernike(zernike_numbers, labels, indices)
-                for (n, m), z in zip(zernike_numbers, zf_l.transpose()):
-                    zf[(n, m)][to_indices] = z
-            #
-            # Form factor
-            #
-            ff = 4.0 * np.pi * objects.areas / mperimeters ** 2
-            #
-            # Feret diameter
-            #
-            min_feret_diameter, max_feret_diameter = \
-                feret_diameter(chulls, chull_counts, objects.indices)
-
+                min_feret_diameter, max_feret_diameter = \
+                    feret_diameter(chulls, chull_counts, objects.indices)
+
+            else:
+                ff = np.zeros(0)
+
+            for f, m in ([(F_AREA, objects.areas),
+                          (F_CENTER_X, mcenter_x),
+                          (F_CENTER_Y, mcenter_y),
+                          (F_CENTER_Z, np.ones_like(mcenter_x)),
+                          (F_EXTENT, mextent),
+                          (F_PERIMETER, mperimeters),
+                          (F_SOLIDITY, msolidity),
+                          (F_FORM_FACTOR, ff),
+                          (F_EULER_NUMBER, euler),
+                          (F_MAXIMUM_RADIUS, max_radius),
+                          (F_MEAN_RADIUS, mean_radius),
+                          (F_MEDIAN_RADIUS, median_radius),
+                          (F_MIN_FERET_DIAMETER, min_feret_diameter),
+                          (F_MAX_FERET_DIAMETER, max_feret_diameter)] +
+                             [(self.get_zernike_name((n, m)), zf[(n, m)])
+                              for n, m in zernike_numbers]):
+                self.record_measurement(workspace, object_name, f, m)
         else:
-            ff = np.zeros(0)
-
-        for f, m in ([(F_AREA, objects.areas),
-                      (F_CENTER_X, mcenter_x),
-                      (F_CENTER_Y, mcenter_y),
-                      (F_EXTENT, mextent),
-                      (F_PERIMETER, mperimeters),
-                      (F_SOLIDITY, msolidity),
-                      (F_FORM_FACTOR, ff),
-                      (F_EULER_NUMBER, euler),
-                      (F_MAXIMUM_RADIUS, max_radius),
-                      (F_MEAN_RADIUS, mean_radius),
-                      (F_MEDIAN_RADIUS, median_radius),
-                      (F_MIN_FERET_DIAMETER, min_feret_diameter),
-                      (F_MAX_FERET_DIAMETER, max_feret_diameter)] +
-                         [(self.get_zernike_name((n, m)), zf[(n, m)])
-                          for n, m in zernike_numbers]):
-            self.record_measurement(workspace, object_name, f, m)
+            labels = objects.segmented
+
+            props = skimage.measure.regionprops(labels)
+
+            # Area
+            areas = [prop.area for prop in props]
```

Comment:
We should consolidate similar accumulations (e.g. `areas`, `extents`, and `centers`) into one iterator.